### PR TITLE
Add Storybook

### PR DESCRIPTION
**Pull request by evexoio**
Implements the Storybook part in #58

TODO:

- [x] add more components
- [x] add pre-commit hook to run eslint on js code
- [x] use Storybook knobs for components state management

dec-04-2018 13-59-36

### CODE REVIEW

**Comment by evexoio** [comment]
Added Storybook Knobs
knobs

**Comment by trezy** [comment]
@evexoio: Sorry to keep asking for changes, but instead of a list of classes, can you make them checkboxes? It’d be nice to, for example, see what a table looks like when it’s both bordered and centered. 😁

**Comment by evexoio** [comment]
@trezy Sure, we need to set some boundaries in some components tho,

for example a centered table without borders look ugly and I don't think it's a valid use case, why should we be able to see it?

**Comment by trezy** [comment]
@evexoio: I’d say that’s a valid state. It may not be a state we advertise in future docs, but it’s still a state that a user could implement, possibly with their own styles on top of it. At the very least, I think being able to see it and test it for regressions is still valuable.

**Comment by evexoio** [comment]
@trezy I don't agree fully, but I'll commit anyways. we can always change it back 🙌

File: package.json [hunk 1]
```diff
@@ -4,14 +4,16 @@
...
     "stylelint": "stylelint scss/**/*.scss",
     "build:stylelint": "npm run stylelint -- --fix",
     "build:clean": "rimraf css",
     "build:sass": "node-sass --output-style expanded --source-map true scss/nes.scss css/nes.css",
     "build:autoprefix": "postcss --use autoprefixer --map false --output css/nes.css css/nes.css",
     "build:cleancss": "cleancss -o css/nes.min.css css/nes.css",
-    "deploy": "npm run build"
+    "deploy": "npm run build",
+    "storybook": "start-storybook -p 6006",
+    "build-storybook": "build-storybook"
```
Comment (BcRikko) [inline package.json]: I prefer following commands to separate other commands 😍 
What do you think? 🤔 

```suggestion
    "storybook:build": "build-storybook"
```

File: docs/buttons.stories.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+stories.addDecorator(withKnobs);
+
+stories.add('button', () => {
+  const extraClass = select('class', {
+    btn: 'btn',
+    'btn is-primary': 'btn is-primary',
+    'btn is-success': 'btn is-success',
+    'btn is-warning': 'btn is-warning',
+    'btn is-error': 'btn is-error',
+  }, 'btn');
```
Comment (BcRikko) [inline docs/buttons.stories.js]: I think radios are easier to use. 👍 
```js
const extraClass = radios('class', {
      default: '',
      'is-primary': 'is-primary',
      'is-success': 'is-success',
      'is-warning': 'is-warning',
      'is-error': 'is-error',
  }, '');
```
buttons

File: docs/containers.stories.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  withKnobs, boolean,
+} from '@storybook/addon-knobs';
+
+const stories = storiesOf('Containers', module);
+stories.addDecorator(withKnobs);
+
+stories.add('container', () => {
+  const withTitle = boolean('with-title', false) ? 'with-title' : '';
+  const isCenter = boolean('is-center', false) ? 'is-center' : '';
+  const isRight = boolean('is-right', false) ? 'is-right' : '';
```
Comment (BcRikko) [inline docs/containers.stories.js]: `is-center` and `is-right` are exclusive relationships 👐 
so radios is better 😆 
```js
const withTitle = boolean('with-title', false) ? 'with-title' : '';
  const alignment = radios('alignment', {
    default: '',
    'is-center': 'is-center',
    'is-right': 'is-right'
  }, '');
  // ...
  return `<div class="container ${selectedClasses.join(' ')} ${alignment}">
      <p class="title">Container</p>
      <p>Good morning. Thou hast had a good night's sleep, I hope.</p>
  </div>`;
```

File: docs/balloons.stories.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+const stories = storiesOf('Ballons', module);
+stories.addDecorator(withKnobs);
+
+stories.add('balloon', () => {
+  const selectedClass = select('class', {
+    'balloon from-left': 'balloon from-left',
+    'balloon from-right': 'balloon from-right',
+  }, 'balloon from-left');
+  return `<div class="${selectedClass}"> <p>Hello NES.css</p> </div>`;
```
Comment (BcRikko) [inline docs/balloons.stories.js]:
```js
const selectedClass = radios('direction', {
  none: '',
  'from-left': 'from-left',
  'from-right': 'from-right',
}, '');
return `<div class="balloon ${selectedClass}"> <p>Hello NES.css</p> </div>`;
```

File: docs/icons.stories.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    'octocat animate': 'octocat animate',
+    'icon trophy': 'icon trophy',
+    'icon trophy is-medium': 'icon trophy is-medium',
+    'icon trophy is-large': 'icon trophy is-large',
+    'nes-logo': 'nes-logo',
+    'nes-jp-logo': 'nes-jp-logo',
+    'snes-logo': 'snes-logo',
+    'snes-jp-logo': 'snes-jp-logo',
+  }, 'icon twitter');
+  return `<i class="${selectedClass}"></i>`;
```
Comment (BcRikko) [inline docs/icons.stories.js]:
```js
  const selectedClass = select('class', {
    'icon twitter': 'icon twitter',
    'icon facebook': 'icon facebook',
    'icon github': 'icon github',
    'icon youtube': 'icon youtube',
    'icon close': 'icon close',
    'icon trophy': 'icon trophy',
    'octocat animate': 'octocat animate',
    'nes-logo': 'nes-logo',
    'nes-jp-logo': 'nes-jp-logo',
    'snes-logo': 'snes-logo',
    'snes-jp-logo': 'snes-jp-logo',
  }, 'icon twitter');
  const selectedSize = radios('size', {
    default: '',
    'is-medium': 'is-medium',
    'is-large': 'is-large',
  }, '');
  return `<i class="${selectedClass} ${selectedSize}"></i>`;
```

File: docs/inputs.stories.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+      </label>`)
+  .add('input', () => {
+    const selectedClass = select('class', {
+      input: 'input',
+      'input is-success': 'input is-success',
+      'input is-warning': 'input is-warning',
+      'input is-error': 'input is-error',
+    }, 'input');
+
+    return `<input type="text" id="name_field" class="${selectedClass}" placeholder="NES.css">`;
```
Comment (BcRikko) [inline docs/inputs.stories.js]:
```js
  .add('input', () => {
    const selectedClass = select('class', {
      default: '',
      'is-success': 'is-success',
      'is-warning': 'is-warning',
      'is-error': 'is-error',
    }, '');

    return `<input type="text" id="name_field" class="input ${selectedClass}" placeholder="NES.css">`;
  });
```

File: package.json (hunk 1 above)
Comment (BcRikko) [reply package.json]: I suggested that `storybook:build` is good. but `build:storybook` is better 👍

**Comment by BcRikko** [comment]
Thanks 🙇 
LGTM 🎉

**Pull request merged**